next.config.js

```diff
@@ -21,15 +21,43 @@ const nextConfig = {
       "/favicon-:all*(png)",
       "/app-icon.png",
       "/preview.png",
-    ].map((source) => ({
-      source,
-      headers: [
-        {
-          key: "Cache-control",
-          value: "max-age=3600, stale-while-revalidate",
-        },
-      ],
-    }))
+    ]
+      .map((source) => ({
+        source,
+        headers: [
+          {
+            key: "Cache-control",
+            value: "max-age=3600, stale-while-revalidate",
+          },
+        ],
+      }))
+      .concat({
+        source: "/(.*)?",
+        headers: [
+          {
+            key: "X-Frame-Options",
+            value: "SAMEORIGIN",
+          },
+          {
+            key: "X-Content-Type-Options",
+            value: "nosniff",
+          },
+          {
+            key: "Permissions-Policy",
+            value:
+              "camera=(), microphone=(), geolocation=(), browsing-topics=()",
+          },
+          {
+            key: "Referrer-Policy",
+            value: "origin-when-cross-origin",
+          },
+          {
+            key: "Content-Security-Policy",
+            value:
+              "default-src 'self'; child-src 'none'; object-src 'none'; img-src 'self' proxy.joinmastodon.org blob: data:; style-src 'self' 'unsafe-inline'; script-src 'self' 'unsafe-inline' 'unsafe-eval'; connect-src 'self' api.joinmastodon.org; block-all-mixed-content",
```

Comment:
The `script-src` is unusually wide here. The proposed headers are marginally better than nothing, but we probably want something significantly stricter there if we see value in CSP headers at all for this domain.

Reply:
Updated with a stricter script CSP in production. I tested locally and I did not see any failures